Marlin/Marlin_main.cpp

```diff
@@ -2121,28 +2115,18 @@ static void clean_up_after_endstop_or_probe_move() {
       planner.bed_level_matrix.set_to_identity();
     #endif
 
-    #if ENABLED(DELTA)
-      float z_before = current_position[Z_AXIS],         // Current Z
-            z_mm = stepper.get_axis_position_mm(Z_AXIS); // Some tower's current position
-    #endif
-
     do_blocking_move_to_z(-(Z_MAX_LENGTH + 10), Z_PROBE_SPEED_FAST);
     endstops.hit_on_purpose();
-    SET_Z_FROM_STEPPERS();
+    set_current_from_steppers();
```

Comment:
Due to floating-point imprecision I wanted to avoid altering the current position for X and Y whenever possible. By preserving the current values, you can compare before and after and see that they haven't changed at all. Perhaps this is less important for probing. And the imprecision may only be 0.0001. So let me know if you think this concern is valid.